hudi-common/src/main/java/org/apache/hudi/common/util/SerializationUtils.java

```diff
@@ -117,7 +117,9 @@ public Kryo newKryo() {
       // ensure that kryo doesn't fail if classes are not registered with kryo.
       kryo.setRegistrationRequired(false);
       // This would be used for object initialization if nothing else works out.
-      kryo.setInstantiatorStrategy(new Kryo.DefaultInstantiatorStrategy(new StdInstantiatorStrategy()));
+      Kryo.DefaultInstantiatorStrategy instantiatorStrategy = new Kryo.DefaultInstantiatorStrategy();
+      instantiatorStrategy.setFallbackInstantiatorStrategy(new StdInstantiatorStrategy());
+      kryo.setInstantiatorStrategy(instantiatorStrategy);
       // Handle cases where we may have an odd classloader setup like with libjars
```

Comment:
We already add some shade pattern for the kryo in `hudi-flink-bundle` jar in https://github.com/apache/hudi/pull/4251, does that solve your conflicts here ?

Reply:
yes，now we use hudi-flink，and the hudi only needs to modify the code and does not need to use the bundle jar

Reply:
> yes，now we use hudi-flink，and the hudi only needs to modify the code and does not need to use the bundle jar

@cuibo01 can you clarify the situation please? do you mean hudi flink bundle jar will have kryo lib conflicts? we have shaded kryo for every module

https://github.com/apache/hudi/blob/b2a84eed02e52a75b61579398f49ae72eeb01440/pom.xml#L450